server/internal/store/cluster.go

```diff
@@ -77,12 +97,14 @@ func (s *S) ListClusters() ([]*Cluster, error) {
 
 // DeleteCluster deletes a cluster by cluster ID and tenant ID.
 func (s *S) DeleteCluster(clusterID, tenantID string) error {
-	res := s.db.Unscoped().Where("cluster_id = ? AND tenant_id = ?", clusterID, tenantID).Delete(&Cluster{})
-	if err := res.Error; err != nil {
-		return err
-	}
-	if res.RowsAffected == 0 {
-		return gorm.ErrRecordNotFound
-	}
-	return nil
+	return s.db.Transaction(func(tx *gorm.DB) error {
+		res := s.db.Unscoped().Where("cluster_id = ? AND tenant_id = ?", clusterID, tenantID).Delete(&Cluster{})
+		if err := res.Error; err != nil {
+			return err
+		}
+		if res.RowsAffected == 0 {
+			return gorm.ErrRecordNotFound
+		}
+		return s.DeleteClusterComponents(clusterID)
```

Comment:
We need to pass `tx` to this function to run the DB query in the same transaction.